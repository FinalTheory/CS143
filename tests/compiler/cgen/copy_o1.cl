class A {
  hello:String <- "Hello!\n";
  getGreetings():String { hello };
};

class Main inherits IO {

  main():Object {
    let
      a:A <- new A,
      copy_a:A
    in {
      copy_a <- a.copy();
      out_string(copy_a.getGreetings());
    }
  };

};
