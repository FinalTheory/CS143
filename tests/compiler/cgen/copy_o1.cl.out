SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
Hello!
COOL program successfully executed
Stats -- #instructions : 325
         #reads : 67  #writes 52  #branches 64  #other 142
